exchanges/graphcache/src/extras/relayPagination.ts

```diff
@@ -263,7 +271,8 @@ export const relayPagination = (
     const hasCurrentPage = !!ensureKey(
       cache.resolve(entityKey, fieldName, fieldArgs)
     );
-    if (!hasCurrentPage) {
+
+    if (!hasCurrentPage && (fieldArgs.first || fieldArgs.last)) {
```

Comment:
If we aren't asking for any pages we can just return the totalCount, alternatively we could check whether there's a selectionSet containing nodes/edges/...